hypothesis-python/src/hypothesis/extra/redis.py

```diff
@@ -0,0 +1,75 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2020 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from contextlib import contextmanager
+from datetime import timedelta
+from typing import Iterable
+
+from redis import Redis
+
+from hypothesis.database import ExampleDatabase
+from hypothesis.internal.validation import check_type
+
+
+class RedisExampleDatabase(ExampleDatabase):
+    """Store Hypothesis examples as sets in the given :class:`redis.Redis` datastore.
+
+    This is particularly useful for shared databases, as per the recipe
+    for a :class:`~hypothesis.database.MultiplexedDatabase`.
+    If a test has not been run for ``expire_after``, those examples will be allowed
+    to expire.  The default time-to-live persists examples between weekly runs.
+    """
+
+    _prefix: bytes = b"hypothesis-example:"
```

Comment:
I guess prefix would be better to be configured via some init argument. Imagine single redis..well, hypothesis server which maintains multiple teams. Some of them would like to run GC on their examples for the fresh start. That would be a problem to not touch others examples. And that single redis could be actually a common cluster.